src/modules/test/ManagementModule.cpp

```diff
@@ -0,0 +1,514 @@
+    m_modulePath(path),
+    m_handle(nullptr)
+{
+    m_info.lifetime = Lifetime::Undefined;
+    m_info.userAccount= 0;
+}
+
+ManagementModule::~ManagementModule()
+{
+    Unload();
+}
+
+int ManagementModule::Load()
+{
+    int status = 0;
+
+    if (nullptr != m_handle)
+    {
+        return status;
+    }
+
+    void* m_handle = dlopen(m_modulePath.c_str(), RTLD_LAZY);
+    if (nullptr != m_handle)
+    {
+        const std::vector<std::string> symbols = {g_mmiFuncMmiGetInfo, g_mmiFuncMmiOpen, g_mmiFuncMmiClose, g_mmiFuncMmiSet, g_mmiFuncMmiGet, g_mmiFuncMmiFree};
+
+        for (auto &symbol : symbols)
+        {
+            mmi_t funcPtr = (mmi_t)dlsym(m_handle, symbol.c_str());
+            if (nullptr == funcPtr)
+            {
+                ADD_FAILURE() << symbol.c_str() << " not exported by module " << m_modulePath.c_str();
+                status = EINVAL;
+            }
+        }
+
+        if (0 == status)
+        {
+            m_mmiOpen = reinterpret_cast<Mmi_Open>(dlsym(m_handle, g_mmiFuncMmiOpen.c_str()));
+            m_mmiGetInfo = reinterpret_cast<Mmi_GetInfo>(dlsym(m_handle, g_mmiFuncMmiGetInfo.c_str()));
+            m_mmiClose = reinterpret_cast<Mmi_Close>(dlsym(m_handle, g_mmiFuncMmiClose.c_str()));
+            m_mmiSet = reinterpret_cast<Mmi_Set>(dlsym(m_handle, g_mmiFuncMmiSet.c_str()));
+            m_mmiGet = reinterpret_cast<Mmi_Get>(dlsym(m_handle, g_mmiFuncMmiGet.c_str()));
+            m_mmiFree = reinterpret_cast<Mmi_Free>(dlsym(m_handle, g_mmiFuncMmiFree.c_str()));
+
+            MMI_JSON_STRING payload = nullptr;
+            int payloadSizeBytes = 0;
+
+            if (MMI_OK == CallMmiGetInfo("Azure OsConfig", &payload, &payloadSizeBytes))
+            {
+                rapidjson::Document document;
+                if (document.Parse(payload, payloadSizeBytes).HasParseError())
+                {
+                    ADD_FAILURE() << "Failed to parse info JSON for module '" << m_modulePath.c_str() << "'";
+                    status = EINVAL;
+                }
+                else if (0 != Info::Deserialize(document, m_info))
+                {
+                    status = EINVAL;
+                }
+            }
+            else
+            {
+                ADD_FAILURE() << "Failed to get info for module '" << m_modulePath.c_str() << "'";
+                status = EINVAL;
+            }
+        }
+    }
+    else
+    {
+        status = EINVAL;
+    }
+
+    if (0 == status)
+    {
+        std::stringstream ss;
+        ss << "[";
+        if (m_info.components.size() != 0)
+        {
+            std::copy(m_info.components.begin(), m_info.components.end() - 1, std::ostream_iterator<std::string>(ss, ", "));
+            ss << m_info.components.back();
+        }
+        ss << "]";
+
+        OsConfigLogInfo(NULL, "Loaded '%s' module (v%s) from '%s', supported components: %s", m_info.name.c_str(), m_info.version.ToString().c_str(), m_modulePath.c_str(), ss.str().c_str());
```

Comment:
Remove the OSConfig logging and raw console logging from this code, replace with GTest tracing.